rstudio: clean up and refactor

TomaSajt opened the pull request:
The first 2 commits should not cause any rebuilds.
The 3rd commit should only cause `rstudio-server` to rebuild. (Since it now has empty arrays as `desktopItems` and `qtWrapperArgs` instead of them being undeclared)

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TomaSajt committed:
rstudio: format with nixfmt

TomaSajt committed:
rstudio: don't use libsForQt5.callPackage

drupol commented on pkgs/applications/editors/rstudio/default.nix:
```diff
@@ -73,24 +53,39 @@ let
-(if server then stdenv.mkDerivation else mkDerivation)
-  (rec {
-    inherit pname version src RSTUDIO_VERSION_MAJOR RSTUDIO_VERSION_MINOR RSTUDIO_VERSION_PATCH RSTUDIO_VERSION_SUFFIX;
+stdenv.mkDerivation rec {
```
How about using `finalAttrs` pattern ?

TomaSajt replied:
Thanks for the suggestion. This is still WIP so I'll put it into draft.

drupol commented on pkgs/applications/editors/rstudio/default.nix:
```diff
@@ -73,24 +53,39 @@ let
+  src = fetchFromGitHub {
+    owner = "rstudio";
+    repo = "rstudio";
+    rev = "v" + version;
```
```suggestion
    rev = "refs/tags/v${version}";
```

TomaSajt replied:
Will do later, but wanted the first few commits to be NOOPs

b-rodrigues reviewed (approved):
LGTM

TomaSajt committed:
rstudio: clean up by moving values around

TomaSajt committed:
rstudio: refactor patches and deps, use lib.cmake*

b-rodrigues commented:
pinging @ciil @drupol just to make sure this PR didn't fly under your radar

TomaSajt commented:
I see that the rstudio server test is failing. I'm assuming it's not supposed to. Will check it out.

TomaSajt committed:
rstudioWrapper: only symlink share directory if not server

TomaSajt commented:
Turns out rstudioWrapper expected the empty `$out/share` directory of rstudio-server (which I removed during the refactor) to exist. Now only the non-server version symlinks the share directory.
Fixed and rebased on latest master

TomaSajt commented:
Rebased on master.

Is everything that worked before this PR still working?
I haven't checked since I can't really program in R and haven't used this program before.

If everything is OK, is there anything stopping us from merging?

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2175

jbedo reviewed (approved):
Tested fully functional, previous issue of the visual editor being broken is resolved. LGTM.

TomaSajt commented:
Okay, I'll try to merge it in a day if nobody objects to it.
(though, you can merge sooner if you want to, I just don't like self-merging quickly)

Kupac reviewed (approved):
Visual editor is finally working. I used to have problems with the built-in debugger (it didn't show up in the environment browser), which is also fixed in this PR. LGTM!

PS: compilation needs more RAM than previously. Restricting the number of cores helps.

TomaSajt commented:
Next step will be https://github.com/NixOS/nixpkgs/pull/357034, though we need to wait for electron to get cached.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/r-studio-visual-editing-not-working/43800/4
